Domain/SymfonyAclCache.php

```diff
@@ -0,0 +1,250 @@
+        }
+
+        $this->cache = $cache;
+        $this->permissionGrantingStrategy = $permissionGrantingStrategy;
+        $this->prefix = $prefix;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clearCache()
+    {
+        $this->cache->clear($this->prefix);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function evictFromCacheById($aclId)
+    {
+        $lookupKey = $this->getAliasKeyForIdentity($aclId);
+        if (!$this->cache->hasItem($lookupKey)) {
+            return;
+        }
+
+        $cacheItem = $this->cache->getItem($lookupKey);
+        if ($cacheItem->isHit()) {
+            $key = $cacheItem->get();
+            if ($this->cache->hasItem($key)) {
+                $this->cache->deleteItem($key);
+            }
+        }
+
+        $this->cache->deleteItem($lookupKey);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function evictFromCacheByIdentity(ObjectIdentityInterface $oid)
+    {
+        $key = $this->getDataKeyByIdentity($oid);
+        if (!$this->cache->hasItem($key)) {
+            return;
+        }
+
+        $this->cache->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromCacheById($aclId)
+    {
+        $lookupKey = $this->getAliasKeyForIdentity($aclId);
+        if (!$this->cache->hasItem($lookupKey)) {
+            return;
+        }
+
+        $lookupKeyItem = $this->cache->getItem($lookupKey);
+        if (!$lookupKeyItem->isHit()) {
+            return;
+        }
+
+        $key = $lookupKeyItem->get();
+        if (!$this->cache->hasItem($key)) {
+            $this->cache->deleteItem($lookupKey);
+
+            return;
+        }
+
+        $keyItem = $this->cache->getItem($key);
+        if (!$keyItem->isHit()) {
+            $this->cache->deleteItem($lookupKey);
+
+            return;
+        }
+
+        return $this->unserializeAcl($keyItem->get());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromCacheByIdentity(ObjectIdentityInterface $oid)
+    {
+        $key = $this->getDataKeyByIdentity($oid);
+        if (!$this->cache->hasItem($key)) {
+            return;
+        }
+
+        $cacheItem = $this->cache->getItem($key);
+        if (!$cacheItem->isHit()) {
+            return;
+        }
+
+        return $this->unserializeAcl($cacheItem->get());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putInCache(AclInterface $acl)
+    {
+        if (null === $acl->getId()) {
+            throw new \InvalidArgumentException('Transient ACLs cannot be cached.');
+        }
+
+        if (null !== $parentAcl = $acl->getParentAcl()) {
+            $this->putInCache($parentAcl);
+        }
+
+        $key = $this->getDataKeyByIdentity($acl->getObjectIdentity());
+        $objectIdentityItem = $this->cache->getItem($key);
+        $objectIdentityItem->set(serialize($acl));
+
+        $this->cache->save($objectIdentityItem);
```

Comment:
```suggestion
        $this->cache->saveDeferred($objectIdentityItem);
```

Because we're adding multiple items to the cache, we should add them with `safeDeferred()` first and perform a `commit()` at the end of the method.

Reply:
Updated accordingly